### DISCUSSION TITLE
Laplace equation over a 2D square domain with a hole

### DISCUSSION THREAD

**Discussion by wonghiram** [lululxvi/deepxde#1204] [Q&A]
I am training a Laplace equation with PINN. The solution area and boundary conditions of my research are shown in the figure below.

![捕获](https://user-images.githubusercontent.com/101537011/227719782-94f921a0-c28f-42f9-b539-3e78e03af833.PNG)

Below is part of my code.
`outer_geom = dde.geometry.Rectangle([0, 0], [1, 1])
inner_geom = dde.geometry.Rectangle([0.4, 0.4], [0.6, 0.6])
def boundary_outer(x, on_boundary):
    return on_boundary and outer_geom.on_boundary(x)
def boundary_inner(x, on_boundary):
    return on_boundary and inner_geom.on_boundary(x)
geom = outer_geom - inner_geom
bc_inner = dde.icbc.DirichletBC(geom, lambda x: 1, boundary_inner)
bc_outer = dde.icbc.DirichletBC(geom, lambda x: 0, boundary_outer)
bc = [bc_inner, bc_outer]
data = dde.data.PDE(geom, pde_laplace, bc, num_domain=1500, num_boundary=1520, num_test=2500)
net = dde.nn.FNN([2] + [30] * 10 + [1], "tanh", "Glorot uniform")
model = dde.Model(data, net)
loss_weights = [1, 100, 10]
model.compile("adam", lr=0.001, loss_weights=loss_weights)
model.train(epochs=50000)
model.compile("L-BFGS")
losshistory, train_state = model.train()`
But the training results are weird. As shown below, the interior boundary does not satisfy the Dirichlet boundary condition U=1.

![捕获1](https://user-images.githubusercontent.com/101537011/227720049-d8159da2-6bbb-4911-b56c-8a0d96b61686.PNG)

![捕获2](https://user-images.githubusercontent.com/101537011/227720165-99334c0a-25b0-42f1-89b0-f6954c285cd4.PNG)

What is causing this problem and how can I improve the code?

**Comment by lululxvi** [comment]
Your code is hard to read.

**Reply by wonghiram** [reply]
Sorry.
Here is a screenshot of my code.
Can you give me some suggestions?
![image](https://user-images.githubusercontent.com/101537011/229667838-59963a7c-a427-43f8-b98c-479529f0533d.png)

**Reply by lululxvi** [reply]
Try a larger weight for the inner BC loss.
